From dbee2f199640ec296b049801fe79e35c4b3424f6 Mon Sep 17 00:00:00 2001
From: Max Kellermann <max@duempel.org>
Date: Mon, 09 Jul 2012 23:14:43 +0000
Subject: output_init: put the "convert" filter at the end of the list

No, really!  This fixes a regression of commit 74617389, which
changed the order of filter plugins.
---
diff --git a/src/output_init.c b/src/output_init.c
index d2c86bf..c3b808e 100644
--- a/src/output_init.c
+++ b/src/output_init.c
@@ -213,13 +213,6 @@ ao_base_init(struct audio_output *ao,
 	ao->replay_gain_filter = NULL;
 	ao->other_replay_gain_filter = NULL;
 
-	/* the "convert" filter must be the last one in the chain */
-
-	ao->convert_filter = filter_new(&convert_filter_plugin, NULL, NULL);
-	assert(ao->convert_filter != NULL);
-
-	filter_chain_append(ao->filter, ao->convert_filter);
-
 	/* done */
 
 	return true;
@@ -280,6 +273,13 @@ audio_output_setup(struct audio_output *ao, const struct config_param *param,
 		return false;
 	}
 
+	/* the "convert" filter must be the last one in the chain */
+
+	ao->convert_filter = filter_new(&convert_filter_plugin, NULL, NULL);
+	assert(ao->convert_filter != NULL);
+
+	filter_chain_append(ao->filter, ao->convert_filter);
+
 	return true;
 }
 
--
cgit v0.9.0.3-66-g3205
